pkg/networkpolicy/controller.go

```diff
@@ -371,7 +373,7 @@ func (c *Controller) Run(ctx context.Context) error {
 
 	nf, err := nfqueue.Open(&config)
 	if err != nil {
-		klog.Infof("could not open nfqueue socket: %v", err)
+		logger.Info("could not open nfqueue socket", "error", err)
```

Comment:
Logging like this exist at several places. I am unsure how to handle them. A general recommendation is to _not_ log errors if you return them (let the caller decide). And normally `logger.Error(err, ...)` is used for logging errors, but then it will be at a higher log-level. I made a compromise by keeping the log at Info level.

Reply:
A way to keep info about what caused the error is to [wrap](https://pkg.go.dev/errors) them:
```go
return fmt.Errorf("could not open nfqueue socket %w", err)
```
(ref: [recent discussion on slack](https://kubernetes.slack.com/archives/CHGFYJVAN/p1720715380174519))